Index: editor/_source/internals/fcktools_gecko.js
===================================================================
--- editor/_source/internals/fcktools_gecko.js	(revision 5699)
+++ editor/_source/internals/fcktools_gecko.js	(working copy)
@@ -131,7 +131,15 @@
 
 FCKTools.GetScrollPosition = function( relativeWindow )
 {
-	return { X : relativeWindow.pageXOffset, Y : relativeWindow.pageYOffset } ;
+    var x = 0;
+    var y = 0;
+    while ( relativeWindow && typeof relativeWindow.scrollTop == 'number' && typeof relativeWindow.scrollLeft == 'number' )
+    {
+        x += relativeWindow.scrollLeft;
+        y += relativeWindow.scrollTop;
+        relativeWindow = relativeWindow.parentNode;
+    }
+	return { X : x, Y : y } ;
 }
 
 FCKTools.AddEventListener = function( sourceObject, eventName, listener )
Index: editor/_source/classes/fckpanel.js
===================================================================
--- editor/_source/classes/fckpanel.js	(revision 5699)
+++ editor/_source/classes/fckpanel.js	(working copy)
@@ -235,6 +235,11 @@
 			oPos.y -= nPos.y ;
 		}
 
+        // Plus the offsets provided by parent of the master iframe.
+        var oScrollPosition = FCKTools.GetScrollPosition( window.FCK.LinkedField.parentNode ) ;
+        oPos.x += oScrollPosition.X;
+        oPos.y += oScrollPosition.Y;
+
 		if ( this.IsRTL && !this.IsContextMenu )
 			x = ( x * -1 ) ;
 
